src/web/app/components/extension-confirm-modal/extension-confirm-modal.component.ts

```diff
@@ -37,7 +39,67 @@ export class ExtensionConfirmModalComponent {
   @Output()
   confirmExtensionCallbackEvent: EventEmitter<boolean> = new EventEmitter();
 
-  constructor(public activeModal: NgbActiveModal, private tableComparatorService: TableComparatorService) {}
+  constructor(public activeModal: NgbActiveModal,
+              private tableComparatorService: TableComparatorService,
+              private timezoneService: TimezoneService) {}
+
+  studentsColumnsData: ColumnData[] = [];
+  studentsRowsData: SortableTableCellData[][] = [];
+
+  instructorsColumnsData: ColumnData[] = [];
+  instructorsRowsData: SortableTableCellData[][] = [];
+
+  ngOnInit(): void {
+    this.getTableData();
+  }
+
+  ngOnChanges(): void {
+    this.getTableData();
+  }
+
+  private formatTimestamp(timestamp: number, timeZone: string): string {
+    return this.timezoneService.formatToString(timestamp, timeZone, 'D MMM YYYY h:mm A');
+  }
+
+  private getTableData(): void {
+    let deadlineType = '';
+    if (this.isExtendModal()) deadlineType = 'Original Deadline';
+    if (this.isDeleteModal() || this.isSessionDeleteModal()) deadlineType = 'Current Deadline';
+
+    this.studentsColumnsData = [
+      { header: 'Section', sortBy: SortBy.SECTION_NAME },
+      { header: 'Team', sortBy: SortBy.TEAM_NAME },
+      { header: 'Name', sortBy: SortBy.RESPONDENT_NAME },
+      { header: 'Email', sortBy: SortBy.RESPONDENT_EMAIL },
+      { header: deadlineType, sortBy: SortBy.SESSION_END_DATE },
+    ];
+
+    this.instructorsColumnsData = [
+      { header: 'Name', sortBy: SortBy.RESPONDENT_NAME },
+      { header: 'Email', sortBy: SortBy.RESPONDENT_EMAIL },
+      { header: 'Role', sortBy: SortBy.INSTRUCTOR_PERMISSION_ROLE },
+      { header: deadlineType, sortBy: SortBy.SESSION_END_DATE },
+    ];
+
+    for (const student of this.selectedStudents) {
+      this.studentsRowsData.push([
+        { value: student.sectionName },
+        { value: student.teamName },
+        { value: student.name },
+        { value: student.email },
+        { value: this.formatTimestamp(student.extensionDeadline, 'America/New_York') },
+      ]);
+    }
+
+    for (const instructor of this.selectedInstructors) {
+      this.instructorsRowsData.push([
+        { value: instructor.name },
+        { value: instructor.email },
+        { value: instructor.role },
+        { value: this.formatTimestamp(instructor.extensionDeadline, 'America/New_York') },
+      ]);
+    }
+  }
 
   SortBy: typeof SortBy = SortBy;
   SortOrder: typeof SortOrder = SortOrder;
```

Comment:
Since we now use the `sortable-table` component, I believe we can remove many of methods from lines 131-221.